dynamodb/src/main/scala/akka/stream/alpakka/dynamodb/AwsOp.scala

```diff
@@ -4,241 +4,64 @@
-    override val marshaller = describeTimeToLiveM
-    def toOp: DescribeTimeToLive = this
-  }
-
-  implicit class TransactGetItems(val request: TransactGetItemsRequest) extends AwsOp {
-    override type A = TransactGetItemsRequest
-    override type B = TransactGetItemsResult
-    override val handler = transactGetItemsU
-    override val marshaller = transactGetItemsM
-    def toOp: TransactGetItems = this
-  }
-
-  implicit class TransactWriteItems(val request: TransactWriteItemsRequest) extends AwsOp {
-    override type A = TransactWriteItemsRequest
-    override type B = TransactWriteItemsResult
-    override val handler = transactWriteItemsU
-    override val marshaller = transactWriteItemsM
-    def toOp: TransactWriteItems = this
-  }
+import org.reactivestreams.Publisher
+import software.amazon.awssdk.core.async.SdkPublisher
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient
+import software.amazon.awssdk.services.dynamodb.model._
+import software.amazon.awssdk.services.dynamodb.paginators.{ListTablesPublisher, QueryPublisher, ScanPublisher}
 
-  implicit class UpdateTimeToLive(val request: UpdateTimeToLiveRequest) extends AwsOp {
-    override type A = UpdateTimeToLiveRequest
-    override type B = UpdateTimeToLiveResult
-    override val handler = updateTimeToLiveU
-    override val marshaller = updateTimeToLiveM
-    def toOp: UpdateTimeToLive = this
-  }
+import scala.compat.java8.FutureConverters._
+import scala.concurrent.Future
 
-  /** Java Api */
-  def create(r: BatchGetItemRequest): AwsOp = r
-
-  /** Java Api */
-  def create(r: CreateTableRequest): CreateTable = r
-
-  /** Java Api */
-  def create(r: DeleteItemRequest): DeleteItem = r
-
-  /** Java Api */
-  def create(r: DeleteTableRequest): DeleteTable = r
-
-  /** Java Api */
-  def create(r: DescribeLimitsRequest): DescribeLimits = r
-
-  /** Java Api */
-  def create(r: DescribeTableRequest): DescribeTable = r
-
-  /** Java Api */
-  def create(r: UpdateItemRequest): UpdateItem = r
-
-  /** Java Api */
-  def create(r: UpdateTableRequest): UpdateTable = r
-
-  /** Java Api */
-  def create(r: PutItemRequest): PutItem = r
-
-  /** Java Api */
-  def create(r: BatchWriteItemRequest): BatchWriteItem = r
-
-  /** Java Api */
-  def create(r: GetItemRequest): GetItem = r
-
-  /** Java Api */
-  def create(r: ListTablesRequest): ListTables = r
-
-  /** Java Api */
-  def create(r: DescribeTimeToLiveRequest): DescribeTimeToLive = r
-
-  /** Java Api */
-  def create(r: TransactGetItemsRequest): TransactGetItems = r
-
-  /** Java Api */
-  def create(r: TransactWriteItemsRequest): TransactWriteItems = r
-
-  /** Java Api */
-  def create(r: UpdateTimeToLiveRequest): UpdateTimeToLive = r
+/**
+ * Representation on an AWS dynamodb sdk operation
+ * @param sdkExecute function to be executed on the AWS client
+ * @tparam In dynamodb request type
+ * @tparam Out dynamodb response type
+ */
+sealed class AwsOp[In <: DynamoDbRequest, Out <: DynamoDbResponse](
```

Comment:
Since we are changing the structure of the connector quite a bit, it is good time to rename this as well. I think renaming this to `DynamoDbOp` is a better match. WDYT?

Reply:
I also prefer `DynamoDbOp` considering the bounds on the type parameters

Reply:
What about defining the ops in a trait and mixing it in both `scaladsl.DynamoDb` and `javadsl.DynamoDb` so `DynamoDb` is the only thing to be imported by the users?

Reply:
Interesting idea.

But then the `{java,scala}dsl.DynamoDb` namespace would get two groups of quite different members: Akka Stream factories and DynamoDb operations themselves. That might get distracting when using autocomplete in `DynamoDb` where the `single`, `source` and `flow` methods, that are API entry methods, will be scattered accross all DynamoDb operation values.

Reply:
Legit. I'll leave it in `DynamoDbOp`